File: src/servus/apps/elven/auth/validate.go
package auth

import (
	"encoding/json"
	"io"
	"servus/core/external/validator"
)

// request body that user should send to login.
type Body struct {
	Username string
	Password string
	Type     string
}

func (a *Body) Validate(body io.ReadCloser) (isValid bool) {
	isValid = false

	// body.
	if err := json.NewDecoder(body).Decode(a); err != nil {
		return
	}

	// username.
	var username = a.Username
	if validator.IsEmpty(&username) {
		return
	}

	// password.
	var password = a.Password
	if validator.IsEmpty(&password) {
		return
	}

	// auth type.
	var authType = a.Type
	if validator.IsEmpty(&authType) {
		return
	}
	var isAuthType = authType == "cookie" || authType == "direct"
	if !isAuthType {
		return
	}

	isValid = true
	return
}

File: src/elven/src/entities/article/articles/validate.ts
/** article validator */
export default class Validate {

    /** validate id */
    public static id(val: string): boolean {
        return !!val
    }

    /** validate title */
    public static title(val: string): boolean {
        return val.length <= 124
    }

    /** validate content */
    public static content(val: string): boolean {
        return val && val.length <= 256000
    }

}